lib/checkfunctions.cpp

```diff
@@ -683,3 +683,133 @@ void CheckFunctions::copyElisionError(const Token *tok)
                 "Using std::move for returning object by-value from function will affect copy elision optimization."
                 " More: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-return-move-local");
 }
+
+void CheckFunctions::useStandardLibrary()
+{
+    if (!mSettings->severity.isEnabled(Severity::style))
+        return;
+
+    for (const Scope& scope: mTokenizer->getSymbolDatabase()->scopeList) {
+        if (scope.type != Scope::ScopeType::eFor)
+            continue;
+
+        const Token *forToken = scope.classDef;
+        // for ( initToken ; condToken ; stepToken )
+        const Token* initToken = getInitTok(forToken);
+        if (!initToken)
+            continue;
+        const Token* condToken = getCondTok(forToken);
+        if (!condToken)
+            continue;
+        const Token* stepToken = getStepTok(forToken);
+        if (!stepToken)
+            continue;
+
+        // 1. we expect that idx variable will be initialized with 0
+        const Token* idxToken = initToken->astOperand1();
+        const Token* initVal = initToken->astOperand2();
+        if (!idxToken || !initVal || !initVal->hasKnownIntValue() || initVal->getKnownIntValue() != 0)
+            continue;
+        const auto idxVarId = idxToken->varId();
+        if (0 == idxVarId)
+            continue;
+
+        // 2. we expect that idx will be less of some variable
+        if (!condToken->isComparisonOp())
+            continue;
+
+        const auto checkBoundaryType = [](const Token* t, const bool rightOp) -> bool {
+            const auto *const till = rightOp ? ";" : ">"; // valid end of expression
+
+            bool result = nullptr != t;
+            for (; result && nullptr != t && t->str() == till; t = t->next()) {
+                if (t->isNumber()) {
+                    continue;
+                } else if (t->isVariable() && !t->variable()->isArray()) {
+                    continue;
+                } else if (t->isArithmeticalOp()) {
+                    continue;
+                } else if (Token::simpleMatch(t, "sizeof (")) {
+                    t = t->next()->link();
+                    continue;
+                } else if (Token::Match(t, "(|)")) {
+                    continue;
+                } else {
+                    result = false;
+                }
+            }
+            return result && nullptr != t;
+        };
+
+        const auto& secondOp = condToken->str();
+        const bool isLess = "<" == secondOp && checkBoundaryType(condToken->astOperand2(), true) &&
```

Comment:
I have the feeling that `isConstExpression` can be used instead of `checkBoundaryType`

Reply:
For test#14: `i < sizeof(int)*(15 + 42/2 - 7);` it returns `false` unfortunately. I guess I could try to investigate it a little bit.

Reply:
ok then the `isConstExpression` should handle that expression better.